### ISSUE TITLE
Ms. Pac-Man Maze Madness no music audio when entering the first portal of the first level. 

### ISSUE THREAD

**Issue by BParks21** [flyinghead/flycast#786]
Windows 11, 8086K, GTX 1080ti

When entering the first portal of the first level the music cuts out and stops playing. Not supposed to happen. Running the NTSC version of the game. Confirmed clean dump, tested in bin/cue format and chd format.

**Comment by flyinghead** [comment]
Known issue. You can pause and resume the game to work around it.

**Comment by BParks21** [comment]
Oh ok will test that thanks.

**Comment by kihato** [comment]
I found that many games are GD_PAUSE with SPI_CD_PLAY command.
It works fine if I change the code like this.

>gdromv3.cpp

```C++
void libCore_CDDA_Sector(s16* sector)
{
	//silence ! :p
	if (cdda.status == cdda_t::Playing)
	{
	            SecNumber.Status = GD_PLAY;
	            libGDR_ReadSector((u8*)sector,cdda.CurrAddr.FAD,1,2352);
```

```C++
case SPI_CD_PLAY:
		{
			...
                        cdda.status = cdda_t::Playing;
                       //SecNumber.Status = GD_PLAY;
                       SecNumber.Status = GD_PAUSE;
```

**Comment by kihato** [comment]
Unfortunately, the above code seems to be incorrect as a bug was found in some games (Bust-A-Move 4).

**Comment by flyinghead** [comment]
It does fix Ms. Pac-Man though, so that's a good hint at what the problem could be.

**Comment by flyinghead** [comment]
I checked on real hardware and Status=GD_PLAY is set immediately when a CD_PLAY command is issued. So it's probably not the right fix.

**Comment by BParks21** [comment]
Any update on this?

**Comment by flyinghead** [comment]
Fixed on the dev branch

**Comment by kihato** [comment]
Unfortunately, there are some issues.
There is a game that serves as a benchmark when testing these issues.

**Star Gladiator 2**
**Bust-A-Move 4**
**Giga Wing**
...and the music player.

**Giga Wing** and the music player appear to be fine this time.
However, with **Ms. Pac-Man** and **Star Gladiator 2**, the music does not stop when the game is paused.

In **Bust-A-Move 4**, if you press the Start button while the title music is playing, the music stops and the game moves to the title screen; however, it seems the music does not actually stop.
Also, the music for the Puzzle Collection character selection screen does not play.

There was also a problem with the music player.
Pausing and resuming playback causes it to skip to the next track.

**Comment by flyinghead** [comment]
Regressions should be fixed on the dev branch.
However I wouldn't rule out other problems with other games...

**Comment by kihato** [comment]
I checked all the games with JP and US CDDA tracks, and everything seems fine. Thank you for all your hard work. Now, the only one left is **The King of Fighters - Dream Match 1999**!

**Comment by flyinghead** [comment]
Yes, I spent a lot of time on this game and I haven't been able to find an acceptable fix. That is a fix that mirrors what I see on real hardware. More work needed I guess.
Could you share your list of games using CDDA? That would be useful for future work on the subject.

**Comment by kihato** [comment]
It is a CSV file. It lists only the game name and the number of tracks.
The "number of tracks" represents the total count, including data tracks, so the actual number is about three fewer.

I have replaced the file with one that includes the PAL version.
[track04-98.zip](https://github.com/user-attachments/files/32427869/track04-98.zip)

**Comment by flyinghead** [comment]
Thank you.
I also make lists like this to remember which game(s) use a particular Dreamcast feature. It's very useful to check for regressions.
